src/Kafka/Consumer/ConsumerProperties.hs

```diff
@@ -116,8 +116,8 @@ clientId (ClientId cid) =
 -- * 'errorCallback'
 -- * 'logCallback'
 -- * 'statsCallback'
-setCallback :: (KafkaConf -> IO ()) -> ConsumerProperties
-setCallback cb = mempty { cpCallbacks = [cb] }
+setCallback :: Callback -> ConsumerProperties
```

Comment:
Shouldn't we also change the type of `ConsumerProperties.cpCallbacks` to `[Callback]` for API consistency, and deconstruct only where callbacks are passed to internal code?

Reply:
Good idea, done.